cmake/SetupUmpireThirdParty.cmake

```diff
@@ -15,6 +15,24 @@ if (EXISTS ${SHROUD_EXECUTABLE})
   include(${CMAKE_CURRENT_BINARY_DIR}/SetupShroud.cmake)
 endif ()
 
+if (UMPIRE_ENABLE_UMAP)
+  if(NOT DEFINED ENV{UMAP_ROOT})
```

Comment:
I'm not sure that environment variables are the best way to do this.  @davidbeckingsale, to you have any thoughts on this?

Reply:
I think it is okay to use both an ENV variable, and a regular CMake variable for UMAP_DIR. However, it would be best to still try the find_path, and only issue an error if the library cannot be found (rather than erroring just because a variable is undefined).